api/src/services/import-services/critter/import-critters-service.ts

```diff
@@ -0,0 +1,306 @@
+import { merge } from 'lodash';
+import { v4 } from 'uuid';
+import { WorkSheet } from 'xlsx';
+import { IDBConnection } from '../../../database/db';
+import { ApiGeneralError } from '../../../errors/api-error';
+import { CSVConfigUtils } from '../../../utils/csv-utils/csv-config-utils';
+import { validateCSVWorksheet } from '../../../utils/csv-utils/csv-config-validation';
+import {
+  CSVConfig,
+  CSVError,
+  CSVHeaderConfig,
+  CSVRowValidated
+} from '../../../utils/csv-utils/csv-config-validation.interface';
+import {
+  getDescriptionCellValidator,
+  getTsnCellValidator,
+  getWlhIDCellValidator
+} from '../../../utils/csv-utils/csv-header-configs';
+import { getLogger } from '../../../utils/logger';
+import { NestedRecord } from '../../../utils/nested-record';
+import { CritterbaseService, IBulkCreate } from '../../critterbase-service';
+import { DBService } from '../../db-service';
+import { PlatformService } from '../../platform-service';
+import { SurveyCritterService } from '../../survey-critter-service';
+import {
+  getCritterAliasCellValidator,
+  getCritterCollectionUnitCellSetter,
+  getCritterCollectionUnitCellValidator,
+  getCritterSexCellSetter,
+  getCritterSexCellValidator
+} from './critter-header-configs';
+
+const defaultLog = getLogger('services/import/import-critters-service');
+
+// Critter CSV static headers
+export type CritterCSVStaticHeader = 'ITIS_TSN' | 'ALIAS' | 'SEX' | 'WLH_ID' | 'DESCRIPTION';
+
+/**
+ *
+ * ImportCrittersService
+ *
+ * @class ImportCrittersService
+ * @extends DBService
+ *
+ */
+export class ImportCrittersService extends DBService {
+  _config: CSVConfig<CritterCSVStaticHeader>;
+
+  surveyId: number;
+  worksheet: WorkSheet;
+
+  configUtils: CSVConfigUtils<CritterCSVStaticHeader>;
+
+  platformService: PlatformService;
+  critterbaseService: CritterbaseService;
+  surveyCritterService: SurveyCritterService;
+
+  /**
+   * Instantiates an instance of ImportCrittersService
+   *
+   * @param {IDBConnection} connection - Database connection
+   * @param {number} surveyId - Survey identifier
+   */
+  constructor(connection: IDBConnection, worksheet: WorkSheet, surveyId: number) {
+    super(connection);
+
+    this._config = {
+      staticHeadersConfig: {
+        ITIS_TSN: { aliases: ['TAXON', 'SPECIES', 'TSN'] },
+        ALIAS: { aliases: ['NICKNAME', 'NAME', 'ANIMAL_ID'] },
+        SEX: { aliases: ['TEST'] },
+        WLH_ID: { aliases: ['WILDLIFE_HEALTH_ID'], validateCell: getWlhIDCellValidator() },
```

Comment:
We should include "Wildlife Health ID" with spaces and "WLHID" as aliases

Reply:
Updated